src/Traits/Authorizable.php

```diff
@@ -204,4 +207,14 @@ public function doesNotHaveAnyPermissions($permissions)
     {
         return ! $this->hasAnyPermissions($permissions);
     }
+
+    public function getPermissions()
+    {
+        if (! Authorization::$cachesPermissions) {
+            return $this->permissions()->get();
+        }
+        return Cache::remember(Authorization::cacheKey(), Authorization::cacheExpiresIn(), function () {
+            return $this->permissions()->get();
+        });
+    }
```

Comment:
This won't work unfortunately, as this cache will be shared by all users since we're using a single cache key here.

The existing cache is for storing *all* permissions that exist, regardless of user. We'll have to refactor this to support being key'ed by the user ID.

Reply:
This is exactly why I opened the issue. I noted that when checking the permission for a route

```php
$this->authorize('items.create');
```

the default behavior is to retrieve the permissions, via ```$this->permissions()``` by running a where query. 

My idea was to cache all the permissions and use the collection in cache to look for the permission. I was able to reduce the number of database queries for a typical route that check multiple permissions in stages by just avoiding the multiple ```->whereName()``` queries to the database.

PS: I check permissions to decide what to include in [API resources](https://laravel.com/docs/11.x/eloquent-resources#conditional-attributes)